scalding-core/src/main/scala/com/twitter/scalding/RichPipe.scala

```diff
@@ -36,7 +36,23 @@ object RichPipe extends java.io.Serializable {
 
   def getNextName: String = "_pipe_" + nextPipe.incrementAndGet.toString
 
-  def assignName(p: Pipe): Pipe = new Pipe(getNextName, p)
+  private val FormerNameBitLength = 8
+  private val FormerNameBitPattern = "^_pipe_([0-9]+)".r
+  private def getFormerNameBit(p: Pipe): String =
```

Comment:
I like this, but i wonder why not just do:
```scala
def getFormerNameBit(p: Pipe): Option[Int] = p.getName match {
  case FormerNameBitPattern(pipeNumber) => Some(pipeNumber.toInt)
  case _ => None
}
```
Then use it with: `new Pipe(s"$getNextName-" + getFormerNameBit(p).fold(p.getName)(_.toString))`